internal/kvstore/util/util.go

```diff
@@ -0,0 +1,29 @@
+package util
+
+import (
+	"errors"
+)
+
+// CheckKeyAndValue returns an error if k == "" or if v == nil
+func CheckKeyAndValue(k string, v any) error {
+	if err := CheckKey(k); err != nil {
+		return err
+	}
+	return CheckVal(v)
+}
+
+// CheckKey returns an error if k == ""
+func CheckKey(k string) error {
+	if k == "" {
+		return errors.New("The passed key is an empty string, which is invalid")
+	}
+	return nil
+}
+
+// CheckVal returns an error if v == nil
+func CheckVal(v any) error {
+	if v == nil {
```

Comment:
Check that it's a pointer. See above how to catch it in compile time.

Reply:
since the interface is using any, this can't be done.